### ISSUE TITLE
Special characters are double-escaped if the CodeHilite extension is enabled

### ISSUE THREAD

**Issue by theMarix** [Python-Markdown/markdown#746]
I use Python-Markdown with the CodeHilite extension to render code examples in my Pelican-based blog. Upgrading to Python-Markdown 3.0.1 I ran into the issue that now all special characters get double-escaped. Thus, the verbose HTML character entity reference is rendered to the user.

Experimenting I found that the issue was introduced by version 3.0.0. The last good version is 2.6.11.

The following sample document can be used to reproduce the issue:

```
This is what code in the Python REPL looks like:

    :::python
    >>> print('Hello, World!')
    Hello, World!
```

Compiling it on Markdown 2.6.11 with `python -m markdown -o html5 -x codehilite sample.md` gives the expected output

```
<p>This is what code in the Python REPL looks like:</p>
<div class="codehilite"><pre><span></span><span class="o">&gt;&gt;&gt;</span> <span class="k">print</span><span class="p">(</span><span class="s1">&#39;Hello, World!&#39;</span><span class="p">)</span>
<span class="n">Hello</span><span class="p">,</span> <span class="n">World</span><span class="err">!</span>
</pre></div>(venv) marix@eddie:~/python-markdown-test> pip install --no-deps --requirements broken.txt 
```

As you can see there are three `&gt;` at the beginning of the code block, leading to the expected rendering result.

In contrast, compiling on Markdown 3.0.1 with `python -m markdown -o html -x codehilite repl-example.md` gives the following output:

```
<p>This is what code in the Python REPL looks like:</p>
<div class="codehilite"><pre><span></span><span class="o">&amp;</span><span class="n">gt</span><span class="p">;</span><span class="o">&amp;</span><span class="n">gt</span><span class="p">;</span><span class="o">&amp;</span><span class="n">gt</span><span class="p">;</span> <span class="k">print</span><span class="p">(</span><span class="s1">&#39;Hello, World!&#39;</span><span class="p">)</span>
<span class="n">Hello</span><span class="p">,</span> <span class="n">World</span><span class="err">!</span>
```

As you can see the `&gt;` got escaped another time, now being represented by `&amp;`, `gt`, and `;`. Thus the rendered output will start with a literal `&gt;&gt;&gt;` instead of the expected output.

**Comment by waylan** [comment]
This is a duplicate of #725 which has been fixed in #727 and will be available in the next release.

**Comment by theMarix** [comment]
Thanks. I had scanned for open issues but failed to check the closed ones. Sorry for the noise.
